AmericanMedicalAssociation/ama-style-guide-2
# [EWL-4608] SG2 | Create the Event Page

## Ticket(s)

**Jira Ticket**
- [EWL-4608: SG2 | Create the Event Page](https://issues.ama-assn.org/browse/EWL-4608)

## Description
As a user, when viewing event details, I want to see a page that uses the [Two Column Right Template](https://americanmedicalassociation.github.io/ama-style-guide-2/?p=templates-two-column-right), a [Masthead](https://americanmedicalassociation.github.io/ama-style-guide-2/?p=organisms-masthead) in the "Content Top" region, a [Membership with CTA](https://americanmedicalassociation.github.io/ama-style-guide-2/?p=organisms-membership-with-cta) at the top of the "Sidebar Secondary" region , and a [Tools Block](https://americanmedicalassociation.github.io/ama-style-guide-2/?p=molecules-tools-list) in the "Sidebar Secondary" region.

This update creates the twig template, descriptive markdown, and relevant scss required to make the event page along with the backstop test case and reference images.

## To Test

- Pull `feature/EWL-4608-create-event-page` branch
- Run `gulp serve` in ama-style-guide-2/styleguide to ensure a local version of style guide is running.
- Navigate to [Pages > Event](http://localhost:3000/?p=event-page).
- Confirm page has placeholders for header and footer along with a masthead, sidebar membership block, sidebar tools list block, 4-up article teaser, and tags as shown in Zeplin mock up.
- Click on the gear icon in the top right corner of the screen and then click "show pattern info". relevant information about the event page should appear.

## Visual Regressions

When testing via backstop, multiple failures occur from resource page test cases. Bug ticket has been created [EWL-5061 SG2 | Correct Backstop images for Inline Resource Links](https://issues.ama-assn.org/browse/EWL-5061).

Reference images for the event page backstop tests have been included in this PR.


## Relevant Screenshots/GIFs

### Full Event Page from Zeplin
![image](https://user-images.githubusercontent.com/4438120/40008727-a3d148aa-5765-11e8-8b77-5d8ee15356cf.png)


### Event page shell in Style Guide
![image](https://user-images.githubusercontent.com/4438120/40008692-8a1bb684-5765-11e8-8ade-0bd0bbeb4b7d.png)


## Remaining Tasks
- [x] Update glossary to include event page definition.
- [x] Confirm with UX that masthead spacing can stay consistent between all pages with that molecule.
- [x] Correct 4-up teasers so they correctly show metadata information as detailed in ticket [EWL-5062 SG2 | Correct 4-Up Teaser Metadata](https://issues.ama-assn.org/browse/EWL-5062)


## Additional Notes
N/A

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)
